lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskQuerySqlProvider.java

```diff
@@ -117,7 +117,8 @@ public static String queryTaskSummariesDb2() {
         + "s.ACCESS_ID IN "
         + "(<foreach item='item' collection='accessIdIn' separator=',' >#{item}</foreach>) "
         + "and "
-        + "s.WORKBASKET_ID = X.WORKBASKET_ID AND s.perm_read = 1 fetch first 1 rows only"
+        + "s.WORKBASKET_ID = X.WORKBASKET_ID AND s.perm_read = 1 AND s.perm_readtasks = 1"
```

Comment:
For me it would have been sufficient to only check the readtasks permission, to keep the complexity small. But in this case it does not increase the complexity too much, or what do you think?